rmw_connext_dynamic_cpp/src/templates.hpp

```diff
@@ -637,7 +637,7 @@ bool set_value<rosidl_typesupport_introspection_cpp::ROS_TYPE_STRING>(
     size_t array_size =
       set_array_size_and_values<rosidl_typesupport_introspection_cpp::ROS_TYPE_STRING>(
       member, ros_message, ros_values);
-    if (!ros_values) {
+    if (!ros_values && array_size) {
```

Comment:
Yeah, this will logically work because if array_size is not equal to zero it will evaluate to true in this conditional, but I'll change it to > 0 to be more readable.